exla/lib/exla/backend.ex

```diff
@@ -82,6 +82,70 @@ defmodule EXLA.Backend do
     EXLA.DeviceBuffer.deallocate(buffer)
   end
 
+  @impl true
+  def to_pointer(%T{data: %B{buffer: buffer}}, opts \\ []) do
+    opts = Keyword.validate!(opts, mode: :local)
+
+    mode =
+      case opts[:mode] do
+        mode when mode in [:local, :cuda_ipc] ->
+          mode
+
+        mode ->
+          raise ArgumentError, "expected one of :local, :cuda_ipc, got: #{inspect(mode)}"
+      end
+
+    case buffer do
+      %EXLA.DeviceBuffer{} ->
+        :ok
+
+      _ ->
+        raise ArgumentError, "tensor must be allocated via a #{DeviceBuffer}"
+    end
+
+    client = EXLA.Client.fetch!(buffer.client_name)
+
+    case EXLA.NIF.get_buffer_device_pointer(client.ref, buffer.ref, mode) do
+      {:ok, {pointer, _size}} ->
+        {:ok, pointer}
+
+      error ->
+        error
+    end
+  end
+
+  @impl true
+  def from_pointer(pointer, type, dims, opts) do
+    template = Nx.template(dims, type, names: opts[:names])
+
+    opts = Keyword.validate!(opts[:backend_opts] || [], [:client_name, :device_id, mode: :local])
```

Comment:
As commented below, this is mixing backend options (such as client name) with pointer options. We probably should treat them separately.